core/amber/src/main/scala/edu/uci/ics/amber/engine/architecture/messaginglayer/NetworkCommunicationActor.scala

```diff
@@ -99,8 +101,16 @@ class NetworkCommunicationActor extends Actor with LazyLogging {
         replyTo.foreach { actor =>
           actor ! RegisterActorRef(actorID, idToActorRefs(actorID))
         }
+      } else if (parentRef != null) {
+        parentRef ! GetActorRef(actorID, replyTo + self)
       } else {
-        context.parent ! GetActorRef(actorID, replyTo + self)
+        throw new WorkflowRuntimeException(
```

Comment:
again please log it. You can do the following: You can put `throw new Exception` as the errorLogAction in  WorkflowLogger. Then, wheenver you call WorkflowLogger.logerror(), it will first log it to console and then call the errorLogAction (which will throw the exception)